crates/starknet-devnet-server/src/api/http/endpoints/postman.rs

```diff
@@ -1,138 +1,130 @@
-use axum::extract::State;
-use axum::Json;
+use starknet_types::felt::{felt_from_prefixed_hex, TransactionHash};
 use starknet_types::rpc::messaging::{MessageToL1, MessageToL2};
 use starknet_types::rpc::transactions::l1_handler_transaction::L1HandlerTransaction;
 
-use super::extract_optional_json_from_request;
 use crate::api::http::error::HttpApiError;
 use crate::api::http::models::{
     FlushParameters, FlushedMessages, MessageHash, MessagingLoadAddress,
-    PostmanLoadL1MessagingContract, TxHash,
+    PostmanLoadL1MessagingContract,
 };
-use crate::api::http::{HttpApiHandler, HttpApiResult};
+use crate::api::json_rpc::error::{ApiError, StrictRpcResult};
+use crate::api::json_rpc::models::TransactionHashOutput;
+use crate::api::json_rpc::{DevnetResponse, JsonRpcHandler};
 use crate::api::Api;
-
-pub async fn postman_load(
-    State(state): State<HttpApiHandler>,
-    Json(data): Json<PostmanLoadL1MessagingContract>,
-) -> HttpApiResult<Json<MessagingLoadAddress>> {
-    postman_load_impl(&state.api, data).await.map(Json::from)
-}
-
-pub async fn postman_flush(
-    State(state): State<HttpApiHandler>,
-    optional_data: Option<Json<FlushParameters>>,
-) -> HttpApiResult<Json<FlushedMessages>> {
-    postman_flush_impl(&state.api, extract_optional_json_from_request(optional_data))
-        .await
-        .map(Json::from)
-}
-
-pub async fn postman_send_message_to_l2(
-    State(state): State<HttpApiHandler>,
-    Json(message): Json<MessageToL2>,
-) -> HttpApiResult<Json<TxHash>> {
-    postman_send_message_to_l2_impl(&state.api, message).await.map(Json::from)
-}
-
-pub async fn postman_consume_message_from_l2(
-    State(state): State<HttpApiHandler>,
-    Json(message): Json<MessageToL1>,
-) -> HttpApiResult<Json<MessageHash>> {
-    postman_consume_message_from_l2_impl(&state.api, message).await.map(Json::from)
-}
+use crate::rpc_core::error::RpcError;
+use crate::rpc_core::request::RpcMethodCall;
+use crate::rpc_core::response::ResponseResult;
+use crate::rpc_handler::RpcHandler;
 
 pub(crate) async fn postman_load_impl(
     api: &Api,
     data: PostmanLoadL1MessagingContract,
-) -> HttpApiResult<MessagingLoadAddress> {
+) -> StrictRpcResult {
     let mut starknet = api.starknet.lock().await;
+    let messaging_contract_address =
+        starknet.configure_messaging(&data.network_url, data.address.as_deref()).await?;
 
-    let messaging_contract_address = starknet
-        .configure_messaging(&data.network_url, data.address.as_deref())
-        .await
-        .map_err(|e| HttpApiError::MessagingError { msg: e.to_string() })?;
+    Ok(DevnetResponse::MessagingContractAddress(MessagingLoadAddress {
+        messaging_contract_address,
+    })
+    .into())
+}
 
-    Ok(MessagingLoadAddress { messaging_contract_address })
+async fn execute_rpc_tx(
+    rpc_handler: &JsonRpcHandler,
+    rpc_call: RpcMethodCall,
+) -> Result<TransactionHash, HttpApiError> {
```

Comment:
what is the reason this method to return HttpApiError

Reply:
Changed return type to `Result<TransactionHash, ApiError>` in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/572/commits/6f13b28b647e5fd7e8990e93175f11d55c1cfd34 and https://github.com/0xSpaceShard/starknet-devnet-rs/pull/572/commits/9a2f6eaa0044c420da83a19f95a5f89184e438e8